torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -295,142 +295,191 @@ def pipeline_llama_tracer(
     return (stage, model)
 
 
-def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
+def apply_tp(model, world_mesh, parallel_dims, job_config: JobConfig):
```

Comment:
let's only pass in `tp_mesh` instead of `world_mesh`

Reply:
That makes sense. 
What could be even cleaner would be to consolidate all meshes and `parallel_dims` into one structure, like a `parallel_context` (or some other name).

Reply:
Modify ParallelDims object so it has properties for 'world_mesh', pp_mesh, and so forth- and uses caching so it only create one actual mesh?

Reply:
Yeah, doable. Seems like a project to be in its own PR?